src/entry-point-static/generatePageContent.tsx

```diff
@@ -0,0 +1,30 @@
+import fs from "fs";
+import React from "react";
+import ReactDOMServer from "react-dom/server";
+import { StaticRouter } from "react-router-dom";
+import { ServerStyleSheet } from "styled-components";
+
+import App from "../entry-point-client/App";
+
+(global as any).React = React;
+
+export interface PageInfo {
+  title: string;
+  location: string;
+}
+export default function generatePageContent({ title, location }: PageInfo) {
+  let styleSheet = new ServerStyleSheet();
+  let contentHtml = ReactDOMServer.renderToString(
+    styleSheet.collectStyles(
+      <StaticRouter location={location}>
+        <App />
+      </StaticRouter>,
+    ),
+  );
+  const stylesHtml = styleSheet.getStyleTags();
+  let template = fs.readFileSync("dist/index.html", "utf-8");
+  template = template.replace("TITLE GOES HERE", title);
```

Comment:
It might just be me having never seen an app boostrapped quite like this before, but I didn't even realize that these three tokens being replaced here were tokens when I first saw them earlier in the PR. I assumed they were placeholders. It would be worth adding a comment next to them to ensure no one accidentally removes them and breaks this process in confusion.

Reply:
Good point -- addressed!